src/seqn/trans.lisp

```diff
@@ -0,0 +1,399 @@
+(in-package :geb.seqn.trans)
+
+(defun remove0-dom (morphism)
+  (remove 0 (dom morphism)))
+
+;; Add range-n function
+
+(defun range-constraints-dom (domain)
+  (cond ((zerop (length domain))   nil)
+        ((= (length domain) 1)    (list (geb.vampir:range-n
+                                         (vamp:make-constant :const (car domain))
+                                         (vamp:make-wire :var
+                                                         (intern (format nil "x~a" 1))))))
+        (t (cons (geb.vampir:range-n (vamp:make-constant :const (car (last domain)))
+                                (vamp:make-wire :var
+                                                (intern (format nil "x~a" (length domain)))))
+            (range-constraints-dom (butlast
+                                    domain))))))
+
+(defmethod to-circuit ((morphism <seqn>) name)
+  "Turns a SeqN term into a Vamp-IR Gate with the given name
+Note that what is happening is that we look at the domain of the morphism
+and skip 0es, making non-zero entries into wires"
+  (let* ((wire-count (length (dom morphism)))
+         (wires (loop for i from 1 to wire-count
+                      collect (vamp:make-wire :var (intern (format nil "x~a" i)
+                                                           :keyword)))))
```

Comment:
```lisp
MAIN> (mapcar (lambda (i) (intern (format nil "X~a" i) :keyword)) (alexandria:iota 10 :start 1))
(:X1 :X2 :X3 :X4 :X5 :X6 :X7 :X8 :X9 :X10)
```

because you are using `intern` using `"x"` and not `"X"` makes a weird symbol that my vmapir printer already fixes